File: src/main/java/com/trade/securities/adapter/in/web/CommodityTradingInfoController.java
package com.trade.securities.adapter.in.web;

import com.trade.securities.application.port.in.GetCommodityTradingInfoUseCase;
import com.trade.securities.domain.CommodityTradingInfo;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/commodities")
@RequiredArgsConstructor
public class CommodityTradingInfoController {

    private final GetCommodityTradingInfoUseCase getCommodityTradingInfoUseCase;

    @GetMapping("/trading-info/{marketType}")
    public List<CommodityTradingInfo> getCommodityTradingInfo(
            @PathVariable String marketType,
            @RequestParam String basDd) {
        return getCommodityTradingInfoUseCase.getCommodityTradingInfo(marketType, basDd);
    }
}

File: src/main/kotlin/com/trade/securities/adapter/in/web/CommodityTradingInfoController.kt
package com.trade.securities.adapter.`in`.web

import com.trade.securities.application.port.`in`.GetCommodityTradingInfoUseCase
import com.trade.securities.domain.CommodityTradingInfo
import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.PathParam
import jakarta.ws.rs.Produces
import jakarta.ws.rs.QueryParam
import jakarta.ws.rs.core.MediaType

@ApplicationScoped
@Path("/api/commodities")
@Produces(MediaType.APPLICATION_JSON)
class CommodityTradingInfoController(
    private val getCommodityTradingInfoUseCase: GetCommodityTradingInfoUseCase
) {

    @GET
    @Path("/trading-info/{marketType}")
    fun getCommodityTradingInfo(
        @PathParam("marketType") marketType: String,
        @QueryParam("basDd") basDd: String
    ): List<CommodityTradingInfo> {
        return getCommodityTradingInfoUseCase.getCommodityTradingInfo(marketType, basDd)
    }
}
